app/src/renderer/vuex/modules/distribution.js

```diff
@@ -0,0 +1,178 @@
+            state.totalRewards = rewards
+        },
+        setDelegationRewards(state, { validatorAddr, rewards }) {
+            Vue.set(state.rewards, validatorAddr, rewards)
+        },
+        setDistributionParameters(state, parameters) {
+            state.parameters = parameters
+        },
+        setOutstandingRewards(state, outstandingRewards) {
+            state.outstandingRewards = outstandingRewards
+        },
+    }
+    const actions = {
+        reconnected({ state, dispatch, rootState: { session } }) {
+            if (state.loading && session.signedIn) {
+                dispatch(`getTotalRewards`)
+            }
+        },
+        resetSessionData({ rootState }) {
+            rootState.delegation = JSON.parse(JSON.stringify(emptyState))
+        },
+        async getTotalRewards(
+            { state, rootState: { session }, commit }
+        ) {
+            state.loading = true
+            try {
+                const rewardsArray = await node.getDelegatorRewards(session.address)
+                const rewards = coinsToObject(rewardsArray)
+                commit(`setTotalRewards`, rewards)
+                state.error = null
+            } catch (error) {
+                commit(`notifyError`, {
+                    title: `Error getting total rewards`,
+                    body: error.message
+                })
+                Sentry.captureException(error)
+                state.error = error
+            }
+            state.loading = false
+            state.loaded = true
+        },
+        async withdrawAllRewards(
+            { rootState: { wallet }, dispatch },
+            { password, submitType }
+        ) {
+            await dispatch(`sendTx`, {
+                type: `postWithdrawDelegatorRewards`,
+                to: wallet.address,
+                password,
+                submitType
+            })
+            await dispatch(`getTotalRewards`)
+        },
+        async getRewardsFromValidator({ state, rootState: { session }, commit }, validatorAddr) {
+            state.loading = true
+            try {
+                const rewardsArray = await node.getDelegatorRewardsFromValidator(session.address, validatorAddr)
+                const rewards = coinsToObject(rewardsArray)
+                commit(`setDelegationRewards`, { validatorAddr, rewards })
+                state.error = null
+            } catch (error) {
+                commit(`notifyError`, {
+                    title: `Error getting rewards from validator`,
+                    body: error.message
+                })
+                Sentry.captureException(error)
+                state.error = error
+            }
+            state.loading = false
+            state.loaded = true
+        },
+        async withdrawRewardsFromValidator(
+            { rootState: { wallet }, commit, dispatch },
+            { validatorAddr, password, submitType }
+        ) {
+            await dispatch(`sendTx`, {
```

Comment:
not for this issue - but this seems kind of dangerous from an implementation perspective. can we make `sendTx` a more generalized function that takes `type` and other relevant `...args`? having to add `wallet.address` and `password` like this is maybe problematic?

@faboweb @sabau ?

Reply:
we need to refactor this. but we will always need to send the password along like this if we don't store it.

Reply:
and `address` is a parameter of `postWithdrawDelegatorRewardsFromValidator`. but open for ideas.

Reply:
i understand. i'm just proposing that we send a tx with type - but data that exists in getters can be populated elsewhere - instead of in the component.

Reply:
> data that exists in getters can be populated elsewhere

sure